### PR TITLE
add Configwiz

### PR DESCRIPTION
**Pull request by jjackson-s**
**Description:** <Describe what has changed.>
- Project called configwiz that originally lived in core but was never published, and is being moved over to contrib to get it a lite version into users' hands. This project helps users build a yaml configuration file for a pipeline. It is a CLI program that prompts users to input each component that they will like to use, and then walks the user through building that individual component. Outputs a .yaml file that can be used as a configuration file for a pipeline.
- Currently does not work due to a blockage in core. Once a PR that I have submitted gets merged, configwiz will begin to work. 
- Many users have expressed that building a yaml configuration file is a pain point, and this change can be a band-aid solution that can assist users given that it will walk users' through the process of building a configuration file.

**Link to tracking Issue:** <Issue number if applicable>
- No direct link to tracking issue, but an ongoing issue that users have been expressing throughout time but just dealing with.

**Testing:** <Describe what testing was performed and which tests were added.>
- About 60% test coverage for the project.

### CODE REVIEW

File: internal/components/components.go [hunk 1]
```diff
@@ -12,27 +12,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package main
+package components
```
Comment (rmfitzpatrick) [inline internal/components/components.go:15]: Were these changes intentional?  Not sure why any component logic changes would be needed (maybe a local git issue?).

File: internal/components/components.go [hunk 2]
```diff
@@ -153,22 +128,42 @@ func components() (component.Factories, error) {
...
+	}
+	factories.Processors, err = component.MakeProcessorFactoryMap(processors...)
+	if err != nil {
+		return component.Factories{}, err
+	}
+
 	exporters := []component.ExporterFactory{
 		alibabacloudlogserviceexporter.NewFactory(),
-		awsemfexporter.NewFactory(),
+		//awsemfexporter.NewFactory(),
```
Comment (rmfitzpatrick) [inline internal/components/components.go:153]: Important that these changes are only additive and necessary.

File: cmd/configschema/configwiz/configwiz/cli.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
```
Comment (rmfitzpatrick) [inline cmd/configschema/configwiz/configwiz/cli.go:1]: Since there is no configschema in this project I don't think the nested dir is necessary.

File: internal/components/components.go (hunk 1 above)
Comment (rmfitzpatrick) [reply internal/components/components.go:15]: oh right, they're in main...

Comment (jjackson-s) [reply internal/components/components.go:15]: Yes that makes sense, I'll remove it and push it in a separate PR.

File: cmd/configwiz/configwiz/printer.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	"fmt"
+	"strings"
+)
+
+func printLine(s string) {
+	fmt.Print(s)
+}
+
+type indentingPrinter2 struct {
```
Comment (pmcollins) [inline cmd/configwiz/configwiz/printer.go:26]: Can we rename this to `indentingPrinter` and remove the current `indentingPrinter`?

**Comment by jjackson-s** [comment]
Currently fixing merge errors that occured during rebase

**Comment by jjackson-s** [comment]
Will rebase commits up to "move file output functionality."

**Comment by jjackson-s** [comment]
@pmcollins Hey, sorry for getting back to you so late, I've been in midterms and haven't really had a chance to look back at this. Anyways, you can go ahead and upload it on your branch to main, it's ok if I do not have any git history on it I would rather just allow you guys to hurry up and get it published.

**Comment by codeboten** [comment]
Closing in favour of #5247

**Comment by jevon-23** [comment]
> Sounds good @jjackson-s, will push my changes shortly. Good luck in your studies and Go Bears!

Thanks good luck in your future endeavors as well! @pmcollins
